cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/store/CredentialIdentityStore.java

```diff
@@ -93,15 +127,21 @@ public Collection<CredentialIdentityId> listForProfile(StructuredTableContext co
    * @throws IOException If any failure reading from storage occurs.
    */
   public Optional<CredentialIdentity> get(StructuredTableContext context, CredentialIdentityId id)
-      throws IOException {
+      throws CipherException, IOException {
     StructuredTable table = context.getTable(CredentialProviderStore.CREDENTIAL_IDENTITIES);
     Collection<Field<?>> key = Arrays.asList(
         Fields.stringField(CredentialProviderStore.NAMESPACE_FIELD,
             id.getNamespace()),
         Fields.stringField(CredentialProviderStore.IDENTITY_NAME_FIELD,
             id.getName()));
-    return table.read(key).map(row -> GSON.fromJson(row
-        .getString(CredentialProviderStore.IDENTITY_DATA_FIELD), CredentialIdentity.class));
+    Optional<StructuredRow> row = table.read(key);
```

Comment:
Probably worth to have a private method to do the raw read such that you don't need to repeat the same logic here and in the exists method.

Reply:
Done